pkg/scalers/arangodb_scaler.go

```diff
@@ -0,0 +1,276 @@
+	// Authentication parameters for connecting to the database
+	// +required
+	arangoDBAuth *authentication.AuthMeta
+	// Specify the unique arangoDB server ID. Only required if bearer JWT is being used.
+	// +optional
+	serverID string
+
+	// The name of the database to be queried.
+	// +required
+	dbName string
+	// The name of the collection to be queried.
+	// +required
+	collection string
+	// The arangoDB query to be executed.
+	// +required
+	query string
+	// A threshold that is used as targetAverageValue in HPA.
+	// +required
+	queryValue int64
+	// A threshold that is used to check if scaler is active.
+	// +optional
+	activationQueryValue int64
+	// Specify whether to verify the server's certificate chain and host name.
+	// +optional
+	unsafeSsl bool
+	// Specify the max size of the active connection pool.
+	// +optional
+	connectionLimit int64
+
+	// The index of the scaler inside the ScaledObject
+	// +internal
+	scalerIndex int
+}
+
+// NewArangoDBScaler creates a new arangodbScaler
+func NewArangoDBScaler(config *ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %s", err)
+	}
+
+	meta, err := parseArangoDBMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing arangoDB metadata: %s", err)
+	}
+
+	client, err := getNewArangoDBClient(meta)
+	if err != nil {
+		return nil, err
+	}
+
+	return &arangoDBScaler{
+		metricType: metricType,
+		metadata:   meta,
+		client:     client,
+		logger:     InitializeLogger(config, "arangodb_scaler"),
+	}, nil
+}
+
+func getNewArangoDBClient(meta *arangoDBMetadata) (driver.Client, error) {
+	var auth driver.Authentication
+
+	conn, err := http.NewConnection(http.ConnectionConfig{
+		Endpoints: strings.Split(meta.endpoints, ","),
+		TLSConfig: &tls.Config{
+			MinVersion:         tls.VersionTLS13,
+			InsecureSkipVerify: meta.unsafeSsl,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a new http connection, %v", err)
+	}
+
+	if meta.arangoDBAuth.EnableBasicAuth {
+		auth = driver.BasicAuthentication(meta.arangoDBAuth.Username, meta.arangoDBAuth.Password)
+	} else if meta.arangoDBAuth.EnableBearerAuth {
+		hdr, err := jwt.CreateArangodJwtAuthorizationHeader(meta.arangoDBAuth.BearerToken, meta.serverID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create bearer token authorization header, %v", err)
+		}
+		auth = driver.RawAuthentication(hdr)
+	}
+
+	client, err := driver.NewClient(driver.ClientConfig{
+		Connection:     conn,
+		Authentication: auth,
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize a new client, %v", err)
+	}
+
+	return client, nil
+}
+
+func parseArangoDBMetadata(config *ScalerConfig) (*arangoDBMetadata, error) {
+	var err error
+
+	// setting default metadata
+	meta := arangoDBMetadata{}
+
+	// parse metaData from ScaledJob config
+	meta.endpoints, err = GetFromAuthOrMeta(config, "endpoints")
```

Comment:
To avoid eventual inconsistencies, we are moving the code to not change the objects before checking the errors, I know in this case it's not a problem at all, but for the sake of consistency, please use an intermediate variable, something like:

```golang
endpoints, err := GetFromAuthOrMeta(config, "endpoints")
if err != nil {
		return nil, err
	}
meta.endpoints = endpoints
```